### PR TITLE
refactor(egen): Refactored Text embedding api semantic search with scann code according to template guidelines

### PR DESCRIPTION
**Pull request by rohith-egen**
Refactored all the cells according to the template notebook
<br>
List of updates:
1. Steps are re-organized based on the [new template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb).
2. Renamed REGION as LOCATION
3. Removed version for google-cloud-aiplatform import 
4. Included colab enterprise link and generalized code for "colab only" cells
<br><br><br>
**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

### CODE REVIEW

File: notebooks/official/generative_ai/text_embedding_api_semantic_search_with_scann.ipynb [hunk 1]
```diff
@@ -31,40 +31,30 @@
...
+        "      <img src=\"https://cloud.google.com/ml-engine/images/colab-logo-32px.png\" alt=\"Google Colaboratory logo\"><br> Open in Colab\n",
         "    </a>\n",
         "  </td>\n",
-        "  <td>\n",
-        "    <a href=\"https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/generative_ai/text_embedding_api_semantic_search_with_scann.ipynb\">\n",
-        "      <img src=\"https://cloud.google.com/ml-engine/images/github-logo-32px.png\" alt=\"GitHub logo\">\n",
-        "      View on GitHub\n",
+        "  <td style=\"text-align: center\">\n",
+        "    <a href=\"https://console.cloud.google.com/vertex-ai/colab/import/https:%2F%2Fraw.githubusercontent.com%2FGoogleCloudPlatform%2Fvertex-ai-samples%2Fmain%2Fnotebooks%2Fofficial%2Fgenerative_ai%2Ftext_embedding_api_semantic_search_with_scann.ipynb\">\n",
+        "      <img width=\"32px\" src=\"https://lh3.googleusercontent.com/JmcxdQi-qOpctIvWKgPtrzZdJJK-J3sWE1RsfjZNwshCFgE_9fULcNpuXYTilIR2hjwN\" alt=\"Google Cloud Colab Enterprise logo\"><br> Open in Colab Enterprise\n",
```
Comment (katiemn) [inline notebooks/official/generative_ai/text_embedding_api_semantic_search_with_scann.ipynb:42]: Can you please update the Colab Enterprise icon src to reflect the new one from the template?

File: notebooks/official/generative_ai/text_embedding_api_semantic_search_with_scann.ipynb [hunk 2]
```diff
@@ -149,169 +139,128 @@
...
-        "**The following steps are required, regardless of your notebook environment.**\n",
-        "\n",
-        "1. [Select or create a Google Cloud project](https://console.cloud.google.com/cloud-resource-manager). When you first create an account, you get a $300 free credit towards your compute/storage costs.\n",
-        "\n",
-        "2. [Make sure that billing is enabled for your project](https://cloud.google.com/billing/docs/how-to/modify-project).\n",
-        "\n",
-        "3. [Enable the Vertex AI API](https://console.cloud.google.com/flows/enableapi?apiid=aiplatform.googleapis.com).\n",
-        "\n",
-        "4. If you are running this notebook locally, you need to install the [Cloud SDK](https://cloud.google.com/sdk)."
+        "## Getting Started"
```
Comment (katiemn) [inline notebooks/official/generative_ai/text_embedding_api_semantic_search_with_scann.ipynb:145]: Can you change "Getting Started" to "Get Started"?

File: notebooks/official/generative_ai/text_embedding_api_semantic_search_with_scann.ipynb [hunk 3]
```diff
@@ -149,169 +139,128 @@
...
+        "id": "a6ab0747c5e6"
       },
       "source": [
-        "#### Set your project ID\n",
-        "\n",
-        "**If you don't know your project ID**, try the following:\n",
-        "* Run `gcloud config list`.\n",
-        "* Run `gcloud projects list`.\n",
-        "* See the support page: [Locate the project ID](https://support.google.com/googleapi/answer/7014113)"
+        "### Install Vertex AI SDK and other required packages"
```
Comment (katiemn) [inline notebooks/official/generative_ai/text_embedding_api_semantic_search_with_scann.ipynb:154]: Modify to "Install Vertex AI SDK for Python and other required packages"

File: notebooks/official/generative_ai/text_embedding_api_semantic_search_with_scann.ipynb (hunk 1 above)
Comment (kittyabs) [reply notebooks/official/generative_ai/text_embedding_api_semantic_search_with_scann.ipynb:42]: here's the url: 

- https://cloud.google.com/ml-engine/images/colab-enterprise-logo-32px.png

File: notebooks/official/generative_ai/text_embedding_api_semantic_search_with_scann.ipynb [hunk 4]
```diff
@@ -149,169 +139,128 @@
...
       "cell_type": "markdown",
       "metadata": {
-        "id": "74ccc9e52986"
+        "id": "e669f8088ac3"
       },
       "source": [
-        "**1. Vertex AI Workbench**\n",
-        "* Do nothing as you are already authenticated."
+        "<div class=\"alert alert-block alert-warning\">\n",
+        "<b>⚠️ The kernel is going to restart. Please wait until it is finished before continuing to the next step. ⚠️</b>\n",
```
Comment (kittyabs) [inline notebooks/official/generative_ai/text_embedding_api_semantic_search_with_scann.ipynb:205]: deleter "Please" and contract "it is":

- Wait until it's finished

**Review by katiemn — APPROVED** [review]

**Pull request merged**